### PR TITLE
#4491 fix refresh on polling, #4438 wrong selected stage

### LINKED ISSUE #4438: Wrong stage focused on deployment selection of a service
Selecting a deployment of a service in the service screen opens wrong stage.

image (16)

**Environment**
* Keptn Version (`keptn version`): 0.8.4

**Describe the bug**
In case that the deployments have the same name, but are delivered in different sequences, it happens that the wrong stage is routed and the view will not provide the details.

**To Reproduce**

1. Deploy the same image twice, but make sure that the second sequence execution is not overriden the first one in all stages
2. Go to service view and select the deployment and stage of the second sequence

**Expected behavior**
It should focus the stage the user clicked on, or the last stage of the current sequence.

### LINKED ISSUE #4491: Service evaluation screen gets refreshed every time after polling data
When selecting a service and examining the evaluations - after each poll, the data is fetched and the whole view is reloaded.
Symptoms:
* The view is cleared
* The loading indicator shows again after each poll
* There is a re-route to the last stage of the service (so when I examined staging I suddenly am back to production again)

Affected component:
ktb-service-details

In the component the deployment seems to get re-set, the stage re-selected and therefore the weird behavior happens.

### PR DESCRIPTION
**Pull request by Kirdock**
Fixes #4491 
Fixes #4438 

Signed-off-by: Klaus Strießnig <k.striessnig@gmail.com>

### CODE REVIEW

File: bridge/client/app/_views/ktb-service-view/ktb-service-view.component.ts [hunk 1]
```diff
@@ -64,6 +66,32 @@ export class KtbServiceViewComponent implements OnInit, OnDestroy {
       this._changeDetectorRef.markForCheck();
     });
 
+    combineLatest([params$, project$]).pipe(take(1)).subscribe(([params, project]) => {
+      if (params.shkeptncontext && params.serviceName) {
+        const service = project.getServices().find(s => s.serviceName === params.serviceName);
+        const paramDeployment = service.deployments.find(deployment => deployment.shkeptncontext === params.shkeptncontext);
```
Comment (laneli) [inline bridge/client/app/_views/ktb-service-view/ktb-service-view.component.ts:72]: Just a nitpick ... as filter returns an array consider it renaming to changedDeployments

**Review by laneli — CHANGES_REQUESTED** [review]

File: bridge/client/app/_components/ktb-service-details/ktb-service-details.component.ts [hunk 2]
```diff
@@ -40,38 +48,35 @@ export class KtbServiceDetailsComponent implements OnInit, OnDestroy{
...
       }
     }
   }
 
   constructor(private _changeDetectorRef: ChangeDetectorRef, private dataService: DataService, private route: ActivatedRoute, private router: Router, private location: Location, private dialog: MatDialog, private clipboard: ClipboardService) {
-
-  }
-
-  ngOnInit(): void {
     this.route.params.pipe(
```
[the remark this answers is not in the document]
Comment (Kirdock) [reply bridge/client/app/_components/ktb-service-details/ktb-service-details.component.ts:62]: It is available and needed to set the project-parameter before the input-parameters are set

File: bridge/client/app/_components/ktb-service-details/ktb-service-details.component.ts [hunk 3]
```diff
@@ -81,28 +86,25 @@ export class KtbServiceDetailsComponent implements OnInit, OnDestroy{
...
-                this.deployment.getStage(evaluation.getStage()).evaluation = evaluation;
+                deployment.getStage(evaluation.getStage()).evaluation = evaluation;
               }
             }
-
-            if (selectLast || !this.selectedStage) {
-              this.selectLastStage();
-            }
+            this._deployment = deployment;
             this._changeDetectorRef.markForCheck();
```
[the remark this answers is not in the document]
Comment (Kirdock) [reply bridge/client/app/_components/ktb-service-details/ktb-service-details.component.ts:93]: `OnPush` is set, so I would leave it as it is

**Review by laneli — DISMISSED** [review]

**Review by laneli — APPROVED** [review]
Looks good

**Pull request merged**